### PR TITLE
OTT-1007: As a OpenWrap Server user, I should be able to log floor de…

### PR DESCRIPTION
**Pull request by Pubmatic-Supriya-Patil**
…tails with soft floor type

# Description

Please add change description or link to ticket, docs, etc.

# Checklist:

- [ ] PR commit list is unique (rebase/pull with the origin branch to keep master clean).
- [ ] JIRA number is added in the PR title and the commit message.
- [ ] Updated the `header-bidding` repo with appropiate commit id.
- [ ] Documented the new changes.

For Prebid upgrade, refer: https://inside.pubmatic.com:8443/confluence/display/Products/Prebid-server+upgrade

### CODE REVIEW

File: exchange/floors.go [hunk 1]
```diff
@@ -189,3 +189,31 @@ func enforceFloors(r *AuctionRequest, seatBids map[openrtb_ext.BidderName]*entit
 
 	return seatBids, rejectionsErrs
 }
+
+func updateBidExt(bidRequestWrapper *openrtb_ext.RequestWrapper, seatBids map[openrtb_ext.BidderName]*entities.PbsOrtbSeatBid) {
```
Comment (pm-nilesh-chate) [inline exchange/floors.go:193]: Add UT for this function

File: exchange/floors.go [hunk 2]
```diff
@@ -151,6 +150,7 @@ func enforceFloors(r *AuctionRequest, seatBids map[openrtb_ext.BidderName]*entit
 		rejectionsErrs = append(rejectionsErrs, err)
 		return seatBids, rejectionsErrs
 	}
+	updateBidExt(r.BidRequestWrapper, seatBids)
```
Comment (pm-nilesh-chate) [inline exchange/floors.go:153]: Update UT for this function w.r.t to `updateBidExt ` changes.

File: exchange/floors.go [hunk 3]
```diff
@@ -189,3 +189,31 @@ func enforceFloors(r *AuctionRequest, seatBids map[openrtb_ext.BidderName]*entit
...
+				continue
+			}
+
+			reqImpCur := reqImp.BidFloorCur
+			if reqImpCur == "" {
+				reqImpCur = "USD"
+				if bidRequestWrapper.Cur != nil {
+					reqImpCur = bidRequestWrapper.Cur[0]
+				}
+			}
```
Comment (pm-nilesh-chate) [inline exchange/floors.go:215]: This logic contradicts with request parsing logic in header-bidding (`updateImpFloorDefaultCurrency`).
1. Header-bidding sets default request to `USD`
2. OW-PBS sets default request to `request.Cur[0]`
Please address the same and do the needful.

File: exchange/floors.go [hunk 4]
```diff
@@ -189,3 +189,31 @@ func enforceFloors(r *AuctionRequest, seatBids map[openrtb_ext.BidderName]*entit
...
+	for _, seatBid := range seatBids {
+
+		for _, bid := range seatBid.Bids {
+			reqImp, ok := impMap[bid.Bid.ImpID]
+			if !ok {
+				continue
+			}
+
+			reqImpCur := reqImp.BidFloorCur
+			if reqImpCur == "" {
```
Comment (pm-nilesh-chate) [inline exchange/floors.go:210]: request/floor currency is already decided at request parsing. We can omit this duplicate logic

**Review by pm-nilesh-chate — CHANGES_REQUESTED** [review]

File: exchange/floors_test.go [hunk 5]
```diff
@@ -1855,3 +1854,144 @@ func TestUpdateBidExtWithFloors(t *testing.T) {
...
+				BidAdjustment:  false,
+				FloorRule:      "*|*|*",
+				FloorRuleValue: 26.02,
+				FloorValue:     12,
+				FloorCurrency:  "JPY",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
```
Comment (pm-jaydeep-mohite) [inline exchange/floors_test.go:1991]: Remove t.Run(tt.name, func(t *testing.T) {}

**Review by pm-jaydeep-mohite — APPROVED** [review]
LGTM

File: exchange/floors.go (hunk 3 above)
Comment (pm-nilesh-chate) [reply exchange/floors.go:215]: @pm-jaydeep-mohite please address this comment

**Pull request merged**